### PR TITLE
fixes #51 OPENTOK-38169 - adding device selection sample

### LINKED ISSUE #51: Device selection sample
👍 This if you would like to see a sample that shows you how to select devices. It would list available microphones/cameras and allow you to select one and publish. We already have https://github.com/opentok/opentok-hardware-setup.js but a sample here could be a bit simpler.

### PR DESCRIPTION
**Pull request by aullman**
In this sample application we show you how to choose an initial microphone and camera to publish and then how to use the [`setAudioSource()`](https://tokbox.com/developer/sdks/js/reference/Publisher.html#setAudioSource) and [`cycleVideo()`](https://tokbox.com/developer/sdks/js/reference/Publisher.html#cycleVideo) methods to change the Microphone and Camera while you are publishing. This sample just creates a simple standalone Publisher without actually publishing it to a Session. 

>This sample is using the new setAudioSource API which is only available in 2.15.0-beta until 2.15.0 is released.

### CODE REVIEW

**Comment by aullman** [comment]
firefox stable failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
safari stable failed errorShot

**Comment by aullman** [comment]
firefox stable failed errorShot

**Comment by aullman** [comment]
firefox stable failed errorShot

**Comment by aullman** [comment]
safari stable failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

File: Publish-Devices/index.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Publish Different Devices</title>
+    
+    <link rel="stylesheet" href="css/app.css">
+    <script src="https://static.opentok.com/v2.15.0-beta/js/opentok.min.js"></script>
```
Comment (msach22) [inline Publish-Devices/index.html]: Since 2.15 is out, should we change this?

Comment (aullman) [reply Publish-Devices/index.html]: yes we should :) I'll update it now.

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
ie 11 failed errorShot

**Comment by aullman** [comment]
Ah, need to update the projects using npm to use 2.15.0 so they have the new plugin.

**Comment by aullman** [comment]
Oops, we broke the typescript definition in 2.15.0. Fixing it now and will do a patch release with 2.15.1

**Pull request merged**